elements/input.lua

```diff
@@ -209,6 +216,76 @@ uie.add("field", {
         end
     end,
 
+    -- Update the cursor and updates the selected area
+    setCursorIndex = function(self, index, updatesSelection, force)
+        local previousIndex = self.index
+        local clearedSelection, selectionStart, selectionStop = false, -1, -1
+        if updatesSelection ~= false then
+            if force or love.keyboard.isDown("lshift", "rshift") then
```

Comment:
I'm not a huge fan of making `setCursorIndex` responsible for checking the keyboard state. Maybe only check if `force` is nil?